bitbots_localization/src/ObservationModel.cpp

```diff
@@ -4,101 +4,50 @@
                                                      std::shared_ptr<Map> map_goals,
                                                      std::shared_ptr<Map> map_field_boundary,
                                                      std::shared_ptr<Map> map_corners,
                                                      std::shared_ptr<Map> map_t_crossings,
-                                                     std::shared_ptr<Map> map_crosses) // todo config!
+                                                     std::shared_ptr<Map> map_crosses,
+                                                     bl::LocalizationConfig &config)
     : particle_filter::ObservationModel<RobotState>() {
   map_lines_ = map_lines;
   map_goals_ = map_goals;
   map_field_boundary_ = map_field_boundary;
   map_corners_ = map_corners;
   map_t_crossings_ = map_t_crossings;
   map_crosses_ = map_crosses;
+  config_ = config;
   particle_filter::ObservationModel<RobotState>::accumulate_weights_ = true;
 
 }
 
-double RobotPoseObservationModel::measure(const RobotState &state) const {
-
-  std::vector<double> lineRatings;
-  std::vector<double> goalRatings;
-  std::vector<double> fieldBoundaryRatings;
-  std::vector<double> cornersRating;
-  std::vector<double> tCrossingsRating;
-  std::vector<double> crossesRating;
-
-  double particle_weight_lines = 0;
-  double particle_weight_goal = 0;
-  double particle_weight_field_boundary = 0;
-  double particle_weight_corners = 0;
-  double particle_weight_t_crossings = 0;
-  double particle_weight_crosses = 0;
-
-  int number_of_effective_measurements = 0;
-
-  if (!last_measurement_lines_.empty()) {
-    number_of_effective_measurements += 1;
-    lineRatings = map_lines_->Map::provideRating(state, last_measurement_lines_);
-    for (double rating : lineRatings) {
-      particle_weight_lines += rating;
-    }
-    particle_weight_lines = particle_weight_lines /
-        lineRatings.size();
-  }
-
-  if (!last_measurement_goal_.empty()) {
-    number_of_effective_measurements += 1;
-    goalRatings = map_goals_->Map::provideRating(state, last_measurement_goal_);
-    for (double rating : goalRatings) {
-      particle_weight_goal += rating;
-    }
-    particle_weight_goal = particle_weight_goal / goalRatings.size();
-
-  }
-
-  if (!last_measurement_field_boundary_.empty()) {
-    number_of_effective_measurements += 1;
-    fieldBoundaryRatings = map_field_boundary_->Map::provideRating(state, last_measurement_field_boundary_);
-    for (double rating : fieldBoundaryRatings) {
-      particle_weight_field_boundary += rating;
+double RobotPoseObservationModel::calculate_weight_for_class(
+    const RobotState &state, 
+    const std::vector<std::pair<double, double>> &last_messurement) const {
+  double particle_weight_for_class = 0;
+  if (!last_messurement.empty()) {
+    number_of_effective_measurements_ += 1;
+    std::vector<double> ratings = map_lines_->Map::provideRating(state, last_messurement);
```

Comment:
do I see this right and you only compare to the lines map? Then this is a huge problem.

Reply:
+1

Reply:
Oh I see. I fix this.